johnnyfive/confluence.py

```diff
@@ -0,0 +1,274 @@
+# -*- coding: utf-8 -*-
+#
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+#  Created on 23-Sep-2021
+#
+#  @author: tbowers
+
+"""Confluence communication module
+
+Confluence API Documentation:
+        https://atlassian-python-api.readthedocs.io/index.html
+"""
+
+# Built-In Libraries
+from time import sleep
+
+# 3rd Party Libraries
+from atlassian import Confluence
```

Comment:
More of a question - if the Confluence user being authenticated doesn't have permission to do something in a given Space, what happens?

Reply:
In general, Confluence barfs and raises an exception.  I've added specific checking code for space permissions.  Right now, that code also raises an exception, but it's a little more explicit.  Maybe we would want to just kick a warning and keep going?

Reply:
I think a warning makes sense, to at least give something traceable back to a resulting log file.  I like the ```perm_dict``` addition and I think it'll be pretty useful for this and future stuff.

Reply:
Permission Exceptions have been changed to Warnings, and the code will happily continue running without performing the requested Confluence action.